yt_dlp/extractor/niconico.py

```diff
@@ -400,27 +397,23 @@ def _real_extract(self, url):
         except ExtractorError as e:
             try:
                 api_data = self._download_json(
-                    'https://www.nicovideo.jp/api/watch/v3/%s?_frontendId=6&_frontendVersion=0&actionTrackId=AAAAAAAAAA_%d' % (video_id, time_millis()), video_id,
+                    'https://www.nicovideo.jp/api/watch/v3/%s?_frontendId=6&_frontendVersion=0&actionTrackId=AAAAAAAAAA_%d' % (video_id, round(time.time() * 1000)), video_id,
                     note='Downloading API JSON', errnote='Unable to fetch data')['data']
             except (ExtractorError, KeyError):
-                if not isinstance(e.cause, compat_HTTPError):
-                    raise e
-                else:
-                    e = e.cause
-                webpage = e.read().decode('utf-8', 'replace')
+                if not isinstance(getattr(e, 'cause', None), compat_HTTPError):
+                    raise
```

Comment:
you are still just re-raising the KeyError 😕

Reply:
fixed